src/fetchData/makeHttpObjects.ts

```diff
@@ -10,13 +10,19 @@ export default function makeHttpObjects({
   reqMocker: OptionsWithDefaults['req'];
   resMocker: OptionsWithDefaults['res'];
 }) {
+  const req = reqMocker(
+    httpMocks.createRequest({
+      url: route,
+      params: { ...params },
+    })
+  );
+
+  if (req.headers.cookie) {
+    document.cookie = req.headers.cookie;
```

Comment:
Since DOM tests should try to get as closer as possible to a realistic user interaction flow, It occurred to be that an user might want to:

- first, set desired cookies in JSDOM environment
- second, append those cookies in `req.headers.cookie`

Here it seems we're setting cookies in JSDOM for the user, but shouldn't a user prepare the JSDOM environment as expected before firing `getPage`?

Ironically we might also consider the opposite approach (even though it also has downsides): checking for `cookies` in `document` and append the to the `req` object. Probably this behaviour could have side-effects and could be optional.

Reply:
A third option could consist of leaving to the use decision/flexibility and add a few line of documentation about how to deal with `req` and cookies.

Reply:
My thoughts on this and library in general:

I'm thinking about this library on a bit higher level than JSDOM. We are rendering routes and not components, which is related to requests/responses in SSR NextJS.

TBH, I don't think I would consider setting `document.cookie` before rendering a route and expect that to work. Naturally I would expect this is done through the req headers.

But it's hard to argue what the best behaviour is without big usage. Current behaviour just seems most obvious to me ATM. Also, that is the last blocker for me to adopt this in a few larger projects and if everything goes smoothly start promoting this library a bit 😉

Reply:
Extra:

Perhaps current behaviour is okay only for initial page render, and should be adjusted for client navigations.

Reply:
Added an extra test to confirm client side login works (setting cookies on "Login" click)

Reply:
`req.headers.cookie` mock is the cookie string, right?

Reply:
> Perhaps current behaviour is okay only for initial page render, and should be adjusted for client navigations.

The issues I can currently see are:
- DOM cookies set after 1st rendering will be overridden e deleted by `req.headers.cookie`  at the following `getPage` call
- DOM cookies removed after 1st rendering will be recreated at the following `getPage` call

Did I get the context right?